src/cli/assert_node_engine_version.ts

```diff
@@ -0,0 +1,25 @@
+import fs from 'fs'
+import path from 'path'
+import semver from 'semver'
+
+type PackageJSON = {
+  engines: { node: string }
+}
+
+const readActualPackageJSON: () => PackageJSON = () =>
+  JSON.parse(
+    fs
+      .readFileSync(path.resolve(__dirname, '..', '..', 'package.json'))
+      .toString()
+  )
+
+export function assertNodeEngineVersion(
+  currentVersion: string,
+  readPackageJSON: () => PackageJSON = readActualPackageJSON
```

Comment:
Is this second parameter for mocking purpose?
Do we actually need this?

Reply:
Yes, this is used
but is it useful?

Reply:
yep, it's just to de-couple it. It felt wrong to have the function be responsible for reading from the file.